TLM/TLM/UI/SubTools/SpeedLimitsTool.cs

```diff
@@ -322,73 +359,152 @@ public class SpeedLimitsTool : SubTool {
 		}
 
+		/// <summary>
+		/// The window for selecting and applying a speed limit
+		/// </summary>
+		/// <param name="num"></param>
 		private void _guiSpeedLimitsWindow(int num) {
 			GUILayout.BeginHorizontal();
+			GUILayout.FlexibleSpace();
+
+			var oldColor = GUI.color;
+			var allSpeedLimits = SpeedLimit.EnumerateSpeedLimits(SpeedUnit.CurrentlyConfigured);
+			allSpeedLimits.Add(0); // add last item: no limit
 
-			Color oldColor = GUI.color;
-			for (int i = 0; i < SpeedLimitManager.Instance.AvailableSpeedLimits.Count; ++i) {
-				if (curSpeedLimitIndex != i)
+			var showMph = GlobalConfig.Instance.Main.DisplaySpeedLimitsMph;
+			var column = 0u; // break palette to a new line at breakColumn
+			var breakColumn = showMph ? SpeedLimit.BREAK_PALETTE_COLUMN_MPH
+				                  : SpeedLimit.BREAK_PALETTE_COLUMN_KMPH; 
+
+			foreach (var speedLimit in allSpeedLimits) {
+				// Highlight palette item if it is very close to its float speed
+				if (SpeedLimit.NearlyEqual(currentPaletteSpeedLimit, speedLimit)) {
 					GUI.color = Color.gray;
-				float signSize = TrafficManagerTool.AdaptWidth(guiSpeedSignSize);
-				if (GUILayout.Button(TextureResources.SpeedLimitTextures[SpeedLimitManager.Instance.AvailableSpeedLimits[i]], GUILayout.Width(signSize), GUILayout.Height(signSize))) {
-					curSpeedLimitIndex = i;
 				}
+
+				_guiSpeedLimitsWindow_AddButton(showMph, speedLimit);
 				GUI.color = oldColor;
 
-				if (i == 6) {
+				// TODO: This can be calculated from SpeedLimit MPH or KMPH limit constants
+				column++;
+				if (column % breakColumn == 0) {
+					GUILayout.FlexibleSpace();
 					GUILayout.EndHorizontal();
 					GUILayout.BeginHorizontal();
-				}
+					GUILayout.FlexibleSpace();
+				} 
 			}
-
+			GUILayout.FlexibleSpace();
 			GUILayout.EndHorizontal();
 
-			if (GUILayout.Button(Translation.GetString("Default_speed_limits"))) {
+			//---------------------
+			// UI buttons row
+			//---------------------
+			GUILayout.BeginHorizontal();
+			GUILayout.FlexibleSpace();
+			if (GUILayout.Button(Translation.GetString("Default_speed_limits"),
+			                     GUILayout.Width(200))) {
 				TrafficManagerTool.ShowAdvisor(this.GetType().Name + "_Defaults");
 				defaultsWindowVisible = true;
 			}
+			GUILayout.FlexibleSpace();
+			GUILayout.EndHorizontal();
 
+			//---------------------
+			// Checkboxes row
+			//---------------------
+			GUILayout.BeginHorizontal();
+			GUILayout.FlexibleSpace();
 			showLimitsPerLane = GUILayout.Toggle(showLimitsPerLane, Translation.GetString("Show_lane-wise_speed_limits"));
+			GUILayout.FlexibleSpace();
+
+			// Display MPH checkbox, if ticked will save global config
+			var displayMph = GlobalConfig.Instance.Main.DisplaySpeedLimitsMph;
+			displayMph = GUILayout.Toggle(displayMph, Translation.GetString("Display_speed_limits_mph"));
+			if (GlobalConfig.Instance.Main.DisplaySpeedLimitsMph != displayMph) {
+				GlobalConfig.Instance.Main.DisplaySpeedLimitsMph = displayMph;
```

Comment:
```suggestion
				Options.setDisplayInMPH(displayMph);
```
Remove
```cs 
GlobalConfig.Instance.Main.DisplaySpeedLimitsMph = displayMph;
GlobalConfig.WriteConfig();
```

and in `Options.cs` line 1572 insert this:
```cs 
	public static void setDisplayInMPH(bool value) {
	    if (displayMphToggle != null) {
	    	displayMphToggle.isChecked = value;
	    }
	}
```
When you call setDisplayInMPH() it will change `isChecked` which calls `onDisplayMphChanged()` line 730 for you ;)

Reply:
Done, tested